### ISSUE TITLE
windows version support issue

### ISSUE THREAD

**Issue by npbcts** [microsoft/TemplateStudio#4622]
### Describe the bug

I use this template create winui3 app.
This app run successful on windows10(19041), but it can not run on windows 11(22621), i don`t know the reason.

When I debug the solution created on win11, it crashed and the exception is:

```
Unhandled exception. System.TypeInitializationException: The type initializer for 'WinRT.ActivationFactory`1' threw an exception.
 ---> System.Runtime.InteropServices.COMException (0x80040154): ????? (0x80040154 (REGDB_E_CLASSNOTREG))
   at WinRT.BaseActivationFactory..ctor(String typeNamespace, String typeFullName)
   at WinRT.ActivationFactory`1..ctor()
   at WinRT.ActivationFactory`1..cctor()
   --- End of inner exception stack trace ---
   at WinRT.ActivationFactory`1.As(Guid iid)
   at Microsoft.UI.Xaml.Application.Make___objRef_global__Microsoft_UI_Xaml_IApplicationStatics()
   at Microsoft.UI.Xaml.Application.get__objRef_global__Microsoft_UI_Xaml_IApplicationStatics()
   at Microsoft.UI.Xaml.Application.Start(ApplicationInitializationCallback callback)
   at myProject.Program.Main(String[] args) in C:\Users\User\source\repos\myProject\obj\x64\Debug\net6.0-windows10.0.22621.0\App.g.i.cs:line 25

```

The app running environment:

Version: Windows 11 Enterprise Evalutation, 22H2, 22621.1194
Computer: Oracle VM VirtualBox

This is a winui3 project dependencies:

```
<PackageReference Include="CommunityToolkit.Mvvm" Version="8.1.0" />
<PackageReference Include="CommunityToolkit.WinUI.UI.Controls.DataGrid" Version="7.1.2" />
<PackageReference Include="Microsoft.Extensions.Hosting" Version="7.0.0" />
<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.2.230118.102" />
<PackageReference Include="Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.0.9" />
<PackageReference Include="WinUIEx" Version="2.1.0" />
```

It create based on croe Microsoft.NETCore.App and .NET Desktop Runtime:

```
$ dotnet --list-sdks
7.0.102 [C:\Program Files\dotnet\sdk]

$ dotnet --list-runtimes
Microsoft.NETCore.App 6.0.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 6.0.13 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

```

How to fix it?

### To Reproduce

 1. Create solution on windows11(22000 or higher version).
 2. Debug the solution in Visual Studio 2022. 
 3. You will see it launch crash.

### Applies to the following platforms:

- [X] WinUI
- [ ] WPF
- [ ] UWP

### About your setup

 - Visual Studio Version:  Microsoft Visual Studio Community 2022 (x64)
 - Template Studio Wizard Version: 5.21
 - Windows Build: Windows 11 Enterprise Evalutation, 22H2, 22621.1194

**Comment by mikebattista** [comment]
Can you clarify your project configuration? What options did you choose in the wizard?

I just tried with an unpackaged, self-contained app with the version of WinAppSDK you're using, and everything worked as expected.

**Comment by npbcts** [comment]
@mikebattista Hi Mike!

I created app using only MSIX Packaging on win11.
Project configuration is as follows:

xxx.csproj file content:
```
﻿<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net6.0-windows10.0.22621.0</TargetFramework>
    <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
    <RootNamespace>dnppt</RootNamespace>
    <ApplicationIcon>Assets/WindowIcon.ico</ApplicationIcon>
    <ApplicationManifest>app.manifest</ApplicationManifest>
    <Platforms>x86;x64;arm64</Platforms>
    <RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
    <PublishProfile>Properties\PublishProfiles\win10-$(Platform).pubxml</PublishProfile>
	<ImplicitUsings>enable</ImplicitUsings>
	<Nullable>enable</Nullable>
	<UseWinUI>true</UseWinUI>
    <EnableMsixTooling>true</EnableMsixTooling>
    <NeutralLanguage>zh-CN</NeutralLanguage>
    <DefaultLanguage>en-US</DefaultLanguage>
    <PackageIcon>icon_base_256.png</PackageIcon>
    <AssemblyVersion></AssemblyVersion>
    <GenerateAppInstallerFile>False</GenerateAppInstallerFile>
    <AppxAutoIncrementPackageRevision>False</AppxAutoIncrementPackageRevision>
    <AppxSymbolPackageEnabled>True</AppxSymbolPackageEnabled>
    <GenerateTestArtifacts>True</GenerateTestArtifacts>
    <AppxBundle>Always</AppxBundle>
    <AppxBundlePlatforms>x64|arm64</AppxBundlePlatforms>
  </PropertyGroup>
  <ItemGroup>
    <None Remove="Data\InnerTestFile_20221001.json" />
  </ItemGroup>
  <ItemGroup>
    <Content Include="Data\Categorys.json">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="Data\InnerTestFile_20221001.json">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
  </ItemGroup>

  <ItemGroup>
    <Manifest Include="$(ApplicationManifest)" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="CommunityToolkit.Mvvm" Version="8.1.0" />
    <PackageReference Include="CommunityToolkit.WinUI.UI.Controls.DataGrid" Version="7.1.2" />
    <PackageReference Include="Microsoft.Extensions.Hosting" Version="7.0.0" />
    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.2.230118.102" />
    <PackageReference Include="Microsoft.Xaml.Behaviors.WinUI.Managed" Version="2.0.9" />
    <PackageReference Include="WinUIEx" Version="2.1.0" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\dnppt.Core\dnppt.Core.csproj" />
  </ItemGroup>
  <ItemGroup>
    <None Update="appsettings.json">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </None>
    <None Update="Assets\icon_base_256.png">
      <Pack>True</Pack>
      <PackagePath>\</PackagePath>
    </None>
    <Page Update="Styles\RichTextBlock.xaml">
      <Generator>MSBuild:Compile</Generator>
    </Page> 
 
  </ItemGroup>

  <ItemGroup Condition="'$(DisableMsixProjectCapabilityAddedByProject)'!='true' and '$(EnableMsixTooling)'=='true'">
    <ProjectCapability Include="Msix" />
  </ItemGroup>

  <ItemGroup>
    <Folder Include="ForPublish\" />
  </ItemGroup>
  
  <PropertyGroup Condition="'$(DisableHasPackageAndPublishMenuAddedByProject)'!='true' and '$(EnableMsixTooling)'=='true'">
    <HasPackageAndPublishMenu>true</HasPackageAndPublishMenu>
  </PropertyGroup>
  
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x86'">
    <DebugType>full</DebugType>
  </PropertyGroup>
  
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x86'">
    <DebugType>full</DebugType>
  </PropertyGroup>
  
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
    <DebugType>full</DebugType>
  </PropertyGroup>
  
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
    <DebugType>full</DebugType>
  </PropertyGroup>
  
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|arm64'">
    <DebugType>full</DebugType>
  </PropertyGroup>
  
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|arm64'">
    <DebugType>full</DebugType>
  </PropertyGroup>
</Project>
```

package.appmanifest file content:
```
﻿<?xml version="1.0" encoding="utf-8"?>

<Package
  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
  xmlns:genTemplate="http://schemas.microsoft.com/appx/developer/templatestudio"
  xmlns:com="http://schemas.microsoft.com/appx/manifest/com/windows10"
  xmlns:desktop="http://schemas.microsoft.com/appx/manifest/desktop/windows10"
  xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest"
  xmlns:uap4="http://schemas.microsoft.com/appx/manifest/uap/windows10/4"
  xmlns:iot="http://schemas.microsoft.com/appx/manifest/iot/windows10"
  IgnorableNamespaces="uap rescap genTemplate mp uap4 iot">


  <Dependencies>
    <TargetDeviceFamily Name="Windows.Universal" MinVersion="10.0.17763.0" MaxVersionTested="10.0.22621.0" />
    <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.17763.0" MaxVersionTested="10.0.22621.0" />
  </Dependencies>

  <Resources>
    <Resource Language="en-US"/>
  </Resources>

  <Applications>
    <Application Id="App"
      Executable="$targetnametoken$.exe"
      EntryPoint="$targetentrypoint$">
      
    </Application>
  </Applications> 

  <genTemplate:Metadata>
    <genTemplate:Item Name="generator" Value="Template Studio"/>
    <genTemplate:Item Name="wizardVersion" Version="v5.2.1" />
    <genTemplate:Item Name="projectType" Value="MenuBar" />
    <genTemplate:Item Name="framework" Value="MVVMToolkit" />
    <genTemplate:Item Name="platform" Value="WinUI" />
    <genTemplate:Item Name="appmodel" Value="Desktop" />
  </genTemplate:Metadata>


  <Capabilities>
    <rescap:Capability Name="runFullTrust"/>
    <Capability Name="internetClient"/>
  </Capabilities>

</Package>
```

You can see dependencies`s version of TemplateStudio (version 5.2.1) have changed cause I updated them.
I tested creating the project without changing the TemplateStudio dependency version, and the startup crash also occurred.

**Comment by mikebattista** [comment]
@DrusTheAxe does anything jump out at you here from the stack?

**Comment by DrusTheAxe** [comment]
Trying to use WinAppSDK's WinRT types w/o them visible to COM. Because they're not in the package graph as expected?

TL;DR try adding `<WindowsAppSdkBootstrapInitialize>true</WindowsAppSdkBootstrapInitialize>` to your csproj ?

> @DrusTheAxe Howard Kapustein FTE does anything jump out at you here from the stack?

I don't see WindowsPackageType defined. Is it set in some imported props/targets?

WindowsAppSdkBootstrapInitialize isn't defined and OutputType=WinExe so if WindowsPackageType=None we'd get WindowsAppSdkBootstrapInitialize=true and the Bootstrapper's auto-initializer included in the exe.

But I don't see WindowsPackageType defined. If not then it's not None and Microsoft.WindowsAppSDK.BootstrapCommon.targets won't set WindowsAppSdkBootstrapInitialize=true.

You can explicitly set WindowsAppSdkBootstrapInitialize. I suspect that will fix the problem. But I'm curious why the auto-detection logic default logic in Microsoft.WindowsAppSDK.BootstrapCommon.targets isn't kicking in in this case.


>To reproduce: 1. Create solution on windows11(22000 or higher version).

Do you recall exactly which project template you selected? Trying to narrow down why WindowsPackageType isn't defined.

**Comment by mikebattista** [comment]
The repro is with an MSIX app. WindowsPackageType is not required in that case as it defaults to MSIX.

WindowsPackageType not being defined is expected here. We only define it in the Template Studio templates for unpackaged apps (set it to None).

**Comment by DrusTheAxe** [comment]
> The repro is with an MSIX app. WindowsPackageType is not required in that case as it defaults to MSIX.
> 
> WindowsPackageType not being defined is expected here. We only define it in the Template Studio templates for unpackaged apps (set it to None).

OK, it's a packaged app. Where's the `appxmanifest.xml`? Please share its content

**Comment by mikebattista** [comment]
@npbcts could you please share appxmanifest.xml?

**Comment by npbcts** [comment]
@mikebattista 

 In this project, app.manifest file as follow:
```
<?xml version="1.0" encoding="utf-8"?>
<assembly manifestVersion="1.0" xmlns="urn:schemas-microsoft-com:asm.v1">
  <assemblyIdentity version="1.0.0.0" name="dnppt.app"/>
    <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
      <application>
        <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}"/>
      </application>
    </compatibility>
  <application xmlns="urn:schemas-microsoft-com:asm.v3">
    <windowsSettings>
      <dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true/PM</dpiAware>
      <dpiAwareness xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">PerMonitorV2, PerMonitor</dpiAwareness>
    </windowsSettings>
  </application>
</assembly>
```
~~~~~~~~~~~~~~~~~~~~~~~~
I don't know if this issue is due to me running the project in a virtual machine. So, tomorrow i will get a windows11 physical machine and test the operation of this project.

**Comment by npbcts** [comment]
@mikebattista Hi Mike!
I tested this project on a windows11 PC,  as you know, this app of self-contained run successful. But I  run the app  of this project which is download from microsoft store, it launch crash again.

Systeminfo of Windows11 PC  as follow: 
```bash
OS Name: Microsoft Windows 11 Home Edition
OS version: 10.0.22621
OS Manufacturer: Microsoft Corporation
OS Configuration: Standalone Workstation
OS build type: Multiprocessor Free
System Manufacturer: Dell Inc.
System Model: Vostro 3020 SFF
System type: x64-based PC
Processor: 1 processor installed.
                   [01]: Intel64 Family 6 Model 183 Stepping 1 GenuineIntel ~2100 Mhz
BIOS Version: Dell Inc. 1.0.0, 2022/10/6
Windows directory: C:\Windows
Total physical memory: 16,073 MB
Available physical memory: 5,771 MB
Virtual Memory: Maximum: 19,017 MB
Virtual Memory: Available: 3,802 MB
Virtual memory: in use: 15,215 MB
```

Application Error information in Event Viewer as follow:
```
Faulting application name: dnppt.exe, version: 1.0.0.0, timestamp: 0x63cf74d2
Faulting module name: Microsoft.ui.xaml.dll, version: 3.0.0.2301, timestamp: 0x655f8cbf
Exception code: 0xc000027b
Error offset: 0x00000000007cd8bc
Faulting Process ID: 0x0x1F98
Faulting application startup time: 0x0x1D946B31B6F32E2
Faulting Application Path: C:\Program Files\WindowsApps\58925WhaleMan.Pro_1.0.2.0_x64__xa75f8gq3f904\dnppt.exe
Faulting Module Path: C:\Program Files\WindowsApps\Microsoft.WindowsAppRuntime.1.2_2000.747.1945.0_x64__8wekyb3d8bbwe\Microsoft.ui.xaml.dll
Report ID: 57c237eb-4967-4b3b-91d4-c334effcf711
Faulting package full name: 58925WhaleMan.Pro_1.0.2.0_x64__xa75f8gq3f904
Faulting Package Relative Application ID: App
```

```html
[-](https://github.com/microsoft/TemplateStudio/issues/4622#) <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
[-](https://github.com/microsoft/TemplateStudio/issues/4622#) <System>
  <Provider Name="Application Error" Guid="{a0e9b465-b939-57d7-b27d-95d8e925ff57}" /> 
  <EventID>1000</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>100</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x8000000000000000</Keywords> 
  <TimeCreated SystemTime="2023-02-22T11:45:07.7787116Z" /> 
  <EventRecordID>5586</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="9924" ThreadID="25872" /> 
  <Channel>Application</Channel> 
  <Computer>Kent</Computer> 
  <Security UserID="S-1-5-21-2121624480-2277374544-1871541796-1002" /> 
  </System>
[-](https://github.com/microsoft/TemplateStudio/issues/4622#) <EventData>
  <Data Name="AppName">dnppt.exe</Data> 
  <Data Name="AppVersion">1.0.0.0</Data> 
  <Data Name="AppTimeStamp">63cf74d2</Data> 
  <Data Name="ModuleName">Microsoft.ui.xaml.dll</Data> 
  <Data Name="ModuleVersion">3.0.0.2301</Data> 
  <Data Name="ModuleTimeStamp">655f8cbf</Data> 
  <Data Name="ExceptionCode">c000027b</Data> 
  <Data Name="FaultingOffset">00000000007cd8bc</Data> 
  <Data Name="ProcessId">0x1f98</Data> 
  <Data Name="ProcessCreationTime">0x1d946b31b6f32e2</Data> 
  <Data Name="AppPath">C:\Program Files\WindowsApps\58925WhaleMan.Pro_1.0.2.0_x64__xa75f8gq3f904\dnppt.exe</Data> 
  <Data Name="ModulePath">C:\Program Files\WindowsApps\Microsoft.WindowsAppRuntime.1.2_2000.747.1945.0_x64__8wekyb3d8bbwe\Microsoft.ui.xaml.dll</Data> 
  <Data Name="IntegratorReportId">57c237eb-4967-4b3b-91d4-c334effcf711</Data> 
  <Data Name="PackageFullName">58925WhaleMan.Pro_1.0.2.0_x64__xa75f8gq3f904</Data> 
  <Data Name="PackageRelativeAppId">App</Data> 
  </EventData>
  </Event>
```
I used `Microsoft.ui.xaml.dll` and  `Exception code: 0xc000027b` as a keywords to search in google and found this problem `https://github.com/microsoft/microsoft-ui-xaml/issues/2773`, but it doesn't give a solution.
